backend/app/controllers/spree/admin/products_controller.rb

```diff
@@ -138,6 +116,21 @@ def variant_scope
       def updating_variant_property_rules?
         params[:product][:variant_property_rules_attributes].present?
       end
+
+      def render_after_update_error
+        # Stops people submitting blank slugs, causing errors when they try to
+        # update the product again
+        @product.slug = @product.slug_was if @product.slug.blank?
+        render action: 'edit'
+      end
+
+      def check_if_updating_variant_property_rules
```

Comment:
This doesn't look like this is a check, it seems more like a normalization? Perhaps a better name for this method would be something like `normalize_variant_property_rules`?